src/rebar_pkg_resource.erl

```diff
@@ -216,3 +221,20 @@ version_pad([Major, Minor, Patch]) ->
     {list_to_integer(Major), list_to_integer(Minor), list_to_integer(Patch)};
 version_pad([Major, Minor, Patch | _]) ->
     {list_to_integer(Major), list_to_integer(Minor), list_to_integer(Patch)}.
+
+maybe_store_etag_in_cache(false = _UpdateETag, _Path, _ETag) ->
+    ok;
+maybe_store_etag_in_cache(true = _UpdateETag, Path, ETag) ->
+    store_etag_in_cache(Path, ETag).
+
+store_etag_in_cache(Path, ETag) ->
+    ok = file:write_file(etag_path(Path), ETag).
+
+etag_path(Path) ->
```

Comment:
This creates packagename-ver.tar.etag. Note that @ferd suggested package-vsn.etag in comment: https://github.com/erlang/rebar3/issues/1743#issuecomment-380040634